/**
* If not stated otherwise in this file or this component's LICENSE
* file the following copyright and licenses apply:
*
* Copyright 2024 RDK Management
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
* http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
**/

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a1c972c..79338ad 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,7 +168,7 @@ else()
         message("CMAKE_CDM_DRM not set")
 endif()
 
-target_link_libraries (gstaamp aamp ${AAMP_COMMON_DEPENDENCIES} )
+target_link_libraries (gstaamp -L$ENV{EXT_INSTALL_PATH}/lib aamp ${AAMP_COMMON_DEPENDENCIES} )
 
 set(LIBAAMP_DEFINES "${AAMP_DEFINES}")
 
diff --git a/gst_subtec/CMakeLists.txt b/gst_subtec/CMakeLists.txt
index 4d08b0b..9a590f8 100644
--- a/gst_subtec/CMakeLists.txt
+++ b/gst_subtec/CMakeLists.txt
@@ -44,9 +44,9 @@ set(GSTSUBTEC_DEPENDENCIES ${GSTREAMERBASE_LIBRARIES} ${GSTREAMER_LIBRARIES})
 add_library(gstsubtecsink SHARED gstsubtecsink.cpp)
 
 if(CMAKE_PLATFORM_UBUNTU)
-    target_link_libraries(gstsubtecsink ${GSTSUBTEC_DEPENDENCIES} ${CMAKE_INSTALL_PREFIX}/lib/libsubtec.so)
+    target_link_libraries(gstsubtecsink -L$ENV{EXT_INSTALL_PATH}/lib ${GSTSUBTEC_DEPENDENCIES} ${CMAKE_INSTALL_PREFIX}/lib/libsubtec.so)
 else()
-    target_link_libraries(gstsubtecsink ${GSTSUBTEC_DEPENDENCIES} "-lsubtec")
+    target_link_libraries(gstsubtecsink -L$ENV{EXT_INSTALL_PATH}/lib ${GSTSUBTEC_DEPENDENCIES} "-lsubtec")
 endif()
 
 add_library(gstsubtecbin SHARED gstsubtecbin.cpp)
